### PR TITLE
KL for MVGaussians

### PR DESCRIPTION
**Pull request by thomaspinder**
This PR implements the Kullback-Leibler divergence from one multivariate Gaussian distribution to another.

### CODE REVIEW

File: numpyro/distributions/kl.py [hunk 1]
```diff
@@ -199,3 +200,30 @@ def kl_divergence(p, q):
...
+
+    Returns:
+        jnp.DeviceArray: The corresponding KL divergence.
+    """
+    N = p.mean.shape[0]
+    iS1 = jnp.linalg.inv(q.covariance_matrix)
+    diff = q.mean - p.mean
+
+    # kl is made of three terms
+    tr_term = jnp.trace(jnp.matmul(iS1, p.covariance_matrix))
```
Comment (fehiepsi) [inline numpyro/distributions/kl.py:221]: We typically perform computation using `scale_tril` to improve speed, something like
```
from jax.scipy.linalg import solve_triangular
tmp = solve_triangular(q.scale_tril, p.scale_tril, lower=True)
tr_term = jnp.square(tmp.reshape(tmp.shape[:-2] + (-1,))).sum(-1)
```

**Review by fehiepsi — COMMENTED** [review]
It seems that we are facing [this issue](https://stackoverflow.com/questions/73929564/entrypoints-object-has-no-attribute-get-digital-ocean). Could you try to add
```
importlib-metadata<5.0
```
to [setup file](https://github.com/pyro-ppl/numpyro/blob/master/setup.py#L51).

File: test/test_distributions.py [hunk 2]
```diff
@@ -2571,3 +2571,13 @@ def sample_binomial_withp0(key):
         return dist.Binomial(total_count=n, probs=0).sample(key)
 
     jax.vmap(sample_binomial_withp0)(random.split(random.PRNGKey(0), 1))
+
+
+@pytest.mark.parametrize(
+    "shape, truth", [(2, 0.30685281944005505), (5, 0.7671320486001374)], ids=str
```
[the remark this answers is not in the document]
Comment (thomaspinder) [reply test/test_distributions.py:2577]: Sure - happy to. Do you have some guidance on how to achieve this? I'm still quite new to numpyro and its idiosyncrasies.

[the remark this answers is not in the document]
Comment (fehiepsi) [reply test/test_distributions.py:2577]: You can do something like `test_kl_dirichlet_dirichlet` above, something like (please adjust `rtol` accordingly)
```
@pytest.mark.parametrize("shape", [(4,), (2, 3)], ids=str)
def test_kl_mvn_mvn(shape):
    p_loc = np.random.normal(size=shape)
    q_loc = np.random.normal(size=shape)
    p_cov_sqrt = np.random.normal(size=shape + (2 * shape[-1],))
    p_cov = p_cov_sqrt @ np.swapaxes(p_cov_sqrt, -2, -1)
    q_cov_sqrt = np.random.normal(size=shape + (2 * shape[-1],))
    q_cov = q_cov_sqrt @ np.swapaxes(q_cov_sqrt, -2, -1)
    p = dist.MultivariatenNormal(p_loc, p_cov)
    q = dist.MultivariatenNormal(q_loc, q_cov)
    actual = kl_divergence(p, q)
    x = p.sample(random.PRNGKey(0), (10_000,)).copy()
    expected = jnp.mean((p.log_prob(x) - q.log_prob(x)), 0)
    assert_allclose(actual, expected, rtol=0.05)
```

**Comment by fehiepsi** [comment]
@thomaspinder Looking like the PR is in a good shape. Could you add the above test and fix the lint issue? Thanks!

File: setup.py [hunk 3]
```diff
@@ -38,6 +38,7 @@
         "multipledispatch",
         "numpy",
         "tqdm",
+        "importlib-metadata<5.0",
```
Comment (fehiepsi) [inline setup.py:41]: you can revert this change now I think

**Comment by fehiepsi** [comment]
Let's close this PR in favor of https://github.com/pyro-ppl/numpyro/pull/1654.
